rclpy/topics/minimal_publisher/publisher_local_function.py

```diff
@@ -0,0 +1,49 @@
+# Copyright 2016 Open Source Robotics Foundation, Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import rclpy
+
+from std_msgs.msg import String
+
+
+def main(args=None):
+    rclpy.init(args)
+
+    node = rclpy.create_node('minimal_publisher')
+    publisher_ = node.create_publisher(String, 'topic')
+
+    msg = String()
+    i = 0
+
+    def timer_callback():
+        nonlocal i
+        msg.data = 'Hello World: %d' % i
+        i += 1
+        print('Publishing: "%s"' % msg.data)
+        publisher_.publish(msg)
+
+    timer = node.create_timer(0.5, timer_callback)
```

Comment:
I don't think a comment is necessary given that we pass the same value to the python `sleep` function. It should be pretty obvious given that all scripts have equivalent behavior. If doubt there is, the variable name in the API is [timer_period_sec](https://github.com/ros2/rclpy/blob/master/rclpy/rclpy/node.py#L96).
I'm not opposed to it but don't think this is a source of confusion